### PR TITLE
feat(extension): init

### PR DESCRIPTION
**Pull request by keiya01**
Setup the state management architecture. 

I will work on the following things after this PR.
- Setup infrastructure
- Setup storytelling
- GraphQL
- Visualizer around

You can test by the following steps.
1. `yarn dev`
2. Setup Re:Earth for the built-in plugin
3. Set widgets
4. Then you can see the screen as follow.
5. You can count up by these buttons. The `local` means local state, and the `global` means global state, so this state will be shared between these widgets.
6. Also the value is stored to the local storage, so it's preserved even if you reload the browser.(We need to do store some value to local storage in VIEW3.0 project.)

Screenshot 2023-08-10 at 17 17 05

### CODE REVIEW

File: extension/src/shared/sharedAtoms/store.ts
Comment (KaWaite) [inline extension/src/shared/sharedAtoms/store.ts:1]: Just a thought (maybe imposible or not a good idea), but do you think we can setup most of this logic inside reearth itself?
My thought is to make a custom hook file that you can pass the plugin state/shared's names (in this case `__PLATEAUVIEW3_STATES` and `__PLATEAUVIEW3_SHARED`)and would return the functions.
I can feel like there are some strong CONS against it, but haven't thought through it enough. Just an idea that you can think about a bit 🤷🏻

File: extension/src/shared/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const hello = () => alert("SHARED");
```
Comment (KaWaite) [inline extension/src/shared/index.ts:1]: ???

File: extension/src/shared/sharedAtoms/store.ts
Comment (keiya01) [reply extension/src/shared/sharedAtoms/store.ts:1]: I think providing the API for global state from Re:Earth side is too much. I think the API will be same with using window directory🤔

File: extension/src/shared/index.ts (hunk 1 above)
Comment (keiya01) [reply extension/src/shared/index.ts]: Removed lol
https://github.com/eukarya-inc/PLATEAU-VIEW-3.0/pull/11#:~:text=fix%3A%20remove%20unnecessary%20code

**Review by KaWaite — APPROVED** [review]

**Review by airslice — APPROVED** [review]

**Pull request merged**